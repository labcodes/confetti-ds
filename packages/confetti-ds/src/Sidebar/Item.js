import React from "react";
import PropTypes from "prop-types";
import Icon from "../Icon";

export default class Item extends React.Component {
  static propTypes = {
    /** This is the item's label. */
    label: PropTypes.string.isRequired,
    /** Sets the icon related to the Item label. Default state: no icon. */
    icon: PropTypes.string,
    /** Callback action to be executed when the Item is clicked. */
    onClick: PropTypes.func.isRequired,
    /** Defines if the current Item is the active one in the Sidebar. */
    isActive: PropTypes.bool,
  };

  static defaultProps = {
    icon: undefined,
    isActive: false,
  };

  render() {
    const { label, icon, isActive, onClick } = this.props;

    return (
      <button
        type="button"
        className={`lab-narrow-sidebar__item${
          isActive ? " lab-narrow-sidebar__item--active" : ""
        }`}
        onClick={onClick}
      >
        {icon ? (
          <Icon
            type={icon}
            color="mineral-60"
            className="lab-narrow-sidebar__item-icon"
          />
        ) : null}
        <span className="lab-narrow-sidebar__item-label">{label}</span>
      </button>
    );
  }
}
