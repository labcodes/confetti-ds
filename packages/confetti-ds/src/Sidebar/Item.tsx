import React, { SyntheticEvent } from "react";
import Icon from "../Icon";
import { IconTypes } from "../constants";

interface ItemProps {
  /** This is the item's label. */
  label: string,
  /** Sets the icon related to the Item label. Default state: no icon. */
  icon?: IconTypes,
  /** Callback action to be executed when the Item is clicked. */
  onClick: (event?: SyntheticEvent) => any,
  /** Defines if the current Item is the active one in the Sidebar. */
  isActive?: boolean,
};

const Item = ({ label, icon, isActive = false, onClick }: ItemProps) => {
  return (
    <button
      type="button"
      className={`lab-narrow-sidebar__item${
        isActive ? " lab-narrow-sidebar__item--active" : ""
      }`}
      onClick={onClick}
    >
      {icon ? (
        <Icon
          type={icon}
          color="mineral-60"
          className="lab-narrow-sidebar__item-icon"
        />
      ) : null}
      <span className="lab-narrow-sidebar__item-label">{label}</span>
    </button>
  );
};

export default Item;
