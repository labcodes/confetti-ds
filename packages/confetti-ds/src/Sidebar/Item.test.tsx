import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Item from "./Item";

describe("Item", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<Item label="Test item" onClick={() => {}} />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with icon", async () => {
    const renderedComponent = renderer
      .create(<Item label="Test item" onClick={() => {}} icon="StackStar" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with isActive", async () => {
    const renderedComponent = renderer
      .create(<Item label="Test item" isActive onClick={() => {}} />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("calls props.onClick when clicked", async () => {
    const mockOnClick = jest.fn();
    const shallowButton = shallow(
      <Item label="Test item" onClick={mockOnClick} />
    );
    shallowButton.simulate("click");
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });
});
