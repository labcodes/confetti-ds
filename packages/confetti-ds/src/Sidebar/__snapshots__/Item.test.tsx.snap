// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders with base props 1`] = `
<button
  className="lab-narrow-sidebar__item"
  onClick={[Function]}
  type="button"
>
  <span
    className="lab-narrow-sidebar__item-label"
  >
    Test item
  </span>
</button>
`;

exports[`Item renders with icon 1`] = `
<button
  className="lab-narrow-sidebar__item"
  onClick={[Function]}
  type="button"
>
  <span
    className="lab-icon lab-icon--star lab-icon--mineral-60 lab-narrow-sidebar__item-icon"
  />
  <span
    className="lab-narrow-sidebar__item-label"
  >
    Test item
  </span>
</button>
`;

exports[`Item renders with isActive 1`] = `
<button
  className="lab-narrow-sidebar__item lab-narrow-sidebar__item--active"
  onClick={[Function]}
  type="button"
>
  <span
    className="lab-narrow-sidebar__item-label"
  >
    Test item
  </span>
</button>
`;
